week-3/server.js

```diff
@@ -0,0 +1,258 @@
+        "values($1) returning id",[newName ],(error,result)=> {
+            if ((error)== undefined){
+                //res.send("New customer added.");
+                res.json(result.rows[0].id);
+                //res.send("New customer added.");
+            }
+            else {
+            res.status(400).send("wrong request "+error)
+}
+        });
+    });
+    
+    //new POST endpoint `/availability`
+    app.post("/availability" ,function (req,res){
+        const productId=parseInt( req.body.productId);
+        const supplierId=parseInt(req.body.supplierId);
+        const price=parseInt(req.body.price);
+        if (price>0 && productIdArr.includes(productId) && supplierIdArr.includes(supplierId)){
+        pool.query("insert into product_availability "+
+        "values($1,$2,$3) ",[productId,supplierId,price ],(error,result)=> {
+            if ((error)== undefined){
+                res.send("New product_availability added.");
+               // res.json(result.rows[0]);
+                //res.send("New customer added.");
+            }
+            else {
+            res.status(400).send("wrong request "+error)
+}
+        });
+    }
+    else
+    res.status(400).send("the price must be positive and the id of the product and the suppliere must be existed  ")
+    });
+    
+
+    app.post("/customers/:customerId/orders" ,function (req,res){
+        const orderDate= req.body.orderDate;
+        const orderRef=req.body.orderRef;
+        const customerId=parseInt(req.params.customerId);
+        if (customersIdArr.includes(customerId) ){
+        pool.query("insert into orders (order_date,order_reference,customer_id) "+
+        "values($1,$2,$3) ",[orderDate,orderRef,customerId ],(error,result)=> {
+            if ((error)== undefined){
+                res.send("New order added.");
+               // res.json(result.rows[0]);
+                //res.send("New customer added.");
+            }
+            else {
+            res.status(400).send("wrong request "+error)
+}
+        });
+    }
+    else
+    res.status(400).send("The  id of the customer  must be existed  ")
+    });
+    
+
+    //PUT endpoint `/customers/:customerId`
+    app.put("/customers/:customerId" ,function (req,res){
+        const customerId=parseInt( req.params.customerId);
+        const newName=req.body.name;
+        const newCity=req.body.city;
+        const newAdd=req.body.address;
+        const newCountry=req.body.country;
+        if (customersIdArr.includes(customerId) ){
+                    pool.query("update customers set name=$1 ,city =$2,address=$3 ,country=$4"+
+        "where id =$5"
+     ,[newName,newCity ,newAdd,newCountry,customerId],(error)=> {
+            if ((error)== undefined){
+                
+                res.send("customer "+customerId+" updated.");
+            }
+        });
+    }else
+    res.status(400).send("This id "+customerId  +" is not  existed  ")
+    });
```

Comment:
See my previous comment about making a query to the database rather than using an in-memory array.